simulator/modules/fetch/bpu/t/unit_test.cpp

```diff
@@ -226,16 +226,16 @@ TEST_CASE( "Cache Miss")
 
     // Check default cache miss behaviour
     Addr PC = 12;
-    CHECK_FALSE(bp->is_hit(PC));
+    CHECK_FALSE( ( bp->is_hit( PC) && bp->is_taken( PC)));
```

Comment:
Please revise — you allow is_hit to be true, but is_taken to be false and vice versa.
It's better just to have to checks for each condition (it does not violate the AAA rule).